Code/Framework/AzToolsFramework/Tests/Prefab/PrefabAssetFixupTests.cpp

```diff
@@ -0,0 +1,184 @@
+
+            ASSERT_NE(context, nullptr);
+
+            MockAssetComponent::Reflect(context);
+
+            AZ::Data::AssetManager::Instance().RegisterHandler(&m_handler, azrtti_typeid<MockAsset>());
+
+            auto entity = aznew AZ::Entity();
+            auto mockAssetComponent = entity->CreateComponent<MockAssetComponent>();
+
+            mockAssetComponent->m_asset =
+                AZ::Data::Asset<MockAsset>(AZ::Uuid::CreateNull(), AZ::Data::AssetType::CreateNull(), "test.asset");
+
+            auto newInstance = AZ::Interface<PrefabSystemComponentInterface>::Get()->CreatePrefab({ entity }, {}, "test.prefab");
+
+            AZStd::string prefabString;
+            ASSERT_TRUE(m_prefabLoaderInterface->SaveTemplateToString(newInstance->GetTemplateId(), prefabString));
+            m_prefabSystemComponent->RemoveAllTemplates();
+
+            AZ::Outcome<PrefabDom, AZStd::string> readPrefabFileResult = AZ::JsonSerializationUtils::ReadJsonString(prefabString);
+
+            ASSERT_TRUE(readPrefabFileResult.IsSuccess());
+
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                m_assetId, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetIdByPath, "test.asset", azrtti_typeid<MockAsset>(),
+                true); // True to register the asset and generate an AssetId for lookup
+
+            m_prefabDom = readPrefabFileResult.TakeValue();
+        }
+
+        void TearDownEditorFixtureImpl() override
+        {
+            PrefabInstantiateTest::TearDownEditorFixtureImpl();
+
+            AZ::Data::AssetManager::Instance().UnregisterHandler(&m_handler);
+        }
+
+        void CheckInstance(const Instance& instance)
+        {
+            const AZ::Entity* loadedEntity = nullptr;
+            instance.GetConstEntities(
+                [&loadedEntity](const AZ::Entity& entity)
+                {
+                    loadedEntity = &entity;
+
+                    return false;
+                });
+
+            auto loadedComponent = loadedEntity->FindComponent<MockAssetComponent>();
+
+            ASSERT_NE(loadedComponent, nullptr);
+
+            ASSERT_STREQ(loadedComponent->m_asset.GetHint().c_str(), "test.asset");
+            ASSERT_EQ(loadedComponent->m_asset->GetId(), m_assetId);
+        }
+
+        MockAssetHandler m_handler;
+        PrefabDom m_prefabDom;
+        AZ::Data::AssetId m_assetId;
+    };
+
+    TEST_F(PrefabFixupTest, Test_LoadInstanceFromPrefabDom_Overload1)
+    {
+        Instance instance;
+        ASSERT_TRUE(PrefabDomUtils::LoadInstanceFromPrefabDom(instance, m_prefabDom));
+
+        CheckInstance(instance);
+    }
+
+    TEST_F(PrefabFixupTest, Test_LoadInstanceFromPrefabDom_Overload2)
+    {
+        Instance instance;
+        AZStd::vector<AZ::Data::Asset<AZ::Data::AssetData>> referencedAssets;
+        ASSERT_TRUE(PrefabDomUtils::LoadInstanceFromPrefabDom(instance, m_prefabDom, referencedAssets));
```

Comment:
so I guess this is testing the two paths that operate when "Control + G" is pressed?

Reply:
Well, its testing the 1 used by Control + G, the one used by "instantiate proc prefab" context menu, and the 3rd one which isn't used by either of them.